webpack/assets/javascripts/react_app/routes/ForemanSwitcher.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Switch } from 'react-router-dom';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { selectComponentByWeight } from '../components/common/Slot/SlotSelectors';
+
+export const ForemanSwitcher = ({ children, externalRoutes }) => {
+  if (Object.keys(externalRoutes).length) {
+    return <Switch>{externalRoutes.concat(children)}</Switch>;
+  }
+  return <Switch>{children}</Switch>;
+};
+
+// map state to props
+const mapStateToProps = state => ({
+  externalRoutes: selectComponentByWeight('routes'),
+});
+
+// export connected component
+export default connect(mapStateToProps)(ForemanSwitcher);
```

Comment:
Can you please explain why the component should be connected?
I don't find any usage of sate/actions

Reply:
when a plugin's route is registered, the store updates as well.
connect  or `useSelector` observe the store and update the component 
I refactored it to use hooks instead of `connect`